.changeset/tame-lemons-play.md

```diff
@@ -0,0 +1,194 @@
+---
+"@latticexyz/cli": patch
+"@latticexyz/common": major
+"@latticexyz/create-mud": major
+"@latticexyz/recs": patch
+"@latticexyz/store-indexer": patch
+---
+
+Templates and examples now use MUD's new sync packages, all built on top of [viem](https://viem.sh/). This greatly speeds up and stabilizes our networking code and improves types throughout.
+
+These new sync packages come with support for our `recs` package, including `encodeEntity` and `decodeEntity` utilities for composite keys.
+
+If you're using `store-cache` and `useRow`/`useRows`, you should wait to upgrade until we have a suitable replacement for those libraries. We're working on a [sql.js](https://github.com/sql-js/sql.js/)-powered sync module that will replace `store-cache`.
+
+## Migrate existing RECS apps to new sync packages
+
+As you migrate, you may find some features replaced, removed, or not included by default. Please [open an issue](https://github.com/latticexyz/mud/issues/new) and let us know if we missed anything.
+
+1. Add `@latticexyz/store-sync` package to your app's `client` package and make sure `viem` is pinned to version `1.3.1` (otherwise you may get type errors)
+
+2. In your `supportedChains.ts`, replace `foundry` chain with our new `mudFoundry` chain.
+
+   ```diff
+   - import { foundry } from "viem/chains";
+   - import { MUDChain, latticeTestnet } from "@latticexyz/common/chains";
+   + import { MUDChain, latticeTestnet, mudFoundry } from "@latticexyz/common/chains";
+
+   - export const supportedChains: MUDChain[] = [foundry, latticeTestnet];
+   + export const supportedChains: MUDChain[] = [mudFoundry, latticeTestnet];
+   ```
+
+3. In `getNetworkConfig.ts`, we'll remove the return type (to let TS infer it for now), remove now-unused config values, and add the viem `chain` object.
+
+   ```diff
+   - export async function getNetworkConfig(): Promise<NetworkConfig> {
+   + export async function getNetworkConfig() {
+   ```
+
+   ```diff
+     const initialBlockNumber = params.has("initialBlockNumber")
+       ? Number(params.get("initialBlockNumber"))
+   -   : world?.blockNumber ?? -1; // -1 will attempt to find the block number from RPC
+   +   : world?.blockNumber ?? 0n;
```

Comment:
do we currently have some mechanism to look for the `HelloWorld` event? or load it from the `worlds.json` file?

Reply:
I am imagining in the future `syncToRecs` will attempt to find a `HelloStore` (which will also have things like protocol version)